### PR TITLE
ENH: refactor checkAndSetDefaultValues into JacksonSpan.Builder

### PR DESCRIPTION
**Pull request by chenqi0805**
Signed-off-by: Chen <19492223+chenqi0805@users.noreply.github.com>

### Description
* As synced with @cmanning09 offline, this PR refactors checkAndSetDefaultValues from JacksonSpan into JacksonSpan.Builder in order to avoid get calls with JsonPointers which is time consuming in the JacksonSpan constructor:

Screen Shot 2022-02-18 at 3 27 55 PM

* Also added a test case on JacksonEvent as the coverage is a little shy.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]
Thanks! Do you have a comparison chart with the time spent after the changes?

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java [hunk 1]
```diff
@@ -56,8 +56,6 @@ protected JacksonSpan(final Builder builder) {
         super(builder);
 
         checkArgument(this.getMetadata().getEventType().equals("TRACE"), "eventType must be of type Trace");
```
Comment (dinujoh) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/trace/JacksonSpan.java:58]: nit: unrelated to this PR, "TRACE" can be moved to static constant for maintainability reasons.

**Review by dinujoh — APPROVED** [review]

**Review by dinujoh — APPROVED** [review]

**Review by sbayer55 — APPROVED** [review]
Thanks for cleaning up the span event code

**Comment by chenqi0805** [comment]
> Thanks! Do you have a comparison chart with the time spent after the changes?

@dlvenable I run another profiling after this change:
Screen Shot 2022-02-21 at 11 33 05 AM

Now the time spent on this method call shrinks significantly

**Pull request merged**